### PR TITLE
Use the npm package for testing the registry

### PR DESCRIPTION
**Pull request by karissa**
Fix tests for node 8 failing with the registry.
  * use a fixed link to datproject.org registry instead of testing on master.
  * enable tests on travis for node 8
  * remove package-lock.json file until we have a good workflow with it.
  * add package-lock.json to .gitignore
  * update auth tests to not run if server is uninstalled (prints a message on how to enable them)

### CODE REVIEW

**Comment by mafintosh** [comment]
Did you mean to check in the lock thing?

**Comment by karissa** [comment]
The lock is already checked in by @jhand. Npm says you're supposed to, although I'm not still sure entirely what the benefits are

**Comment by ralphtheninja** [comment]
FYI, I was installing dat from source yesterday (using the lock file that's on master right now) and got leveldown installed and I was surprised to see that I couldn't find leveldown anywhere (using npm ls leveldown) but later realized it was due to the lock file being out of date. This PR removes that and a tonne of other modules.

**Comment by ralphtheninja** [comment]
Ps This is not a problem if you do `npm i dat -g`, only if you install from source, e.g. `npm install` or `npm link`

**Comment by mafintosh** [comment]
@karissa @ralphtheninja Cool! (i don't really get that whole lock thing either). Changed travis to run on 8 instead of 7 to see if that fixes that weird error.

**Comment by ralphtheninja** [comment]
Out of curiosity I compared the contents of the installed version of dat (inside ~/.nvm folder) and something caught my eye, see below (don't mind the `v0.33.2` thing, it's the nvm version)

screenshot from 2017-06-09 11-01-43

It seems that the source tree doesn't contain any `.npmignore` but the installed package does (where does it come from?)

**Comment by ralphtheninja** [comment]
@mafintosh I suspect the weird build errors are related to missing prebuilt versions for that version of node so it falls back to build step but fails since gcc is old? (Need at least 4.8 for c++11 things).

**Comment by mafintosh** [comment]
@ralphtheninja Yea, the lock thing is forcing it to use an old version of utp-native (1.5.0 vs 1.5.0). How do I make it not do that?

**Comment by mafintosh** [comment]
I think npm automatically include the npmignore on install but could be mistaking

**Comment by mafintosh** [comment]
I.e all builds are available but package-lock is making it use an old version that do not have them.

> @ralphtheninja Yea, the lock thing is forcing it to use an old version of utp-native (1.5.0 vs 1.5.1). How do I make it not do that?

**Comment by ralphtheninja** [comment]
@mafintosh Try removing package-lock.json

**Comment by ralphtheninja** [comment]
Merging https://github.com/datproject/dat/pull/802 will make travis work again

It's not related to your code but rather `node-sass` not having any precompiled binaries for node 8

**Comment by karissa** [comment]
I'm disabling the need for node-sass, making auth tests optional.

File: test/helpers/auth-server.js [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+  Server = require('dat-registry-api/server')
+  initDb = require('dat-registry-api/server/database/init')
+} catch (e) {
+  console.log('Disabling auth tests, run `npm install dat-registry-api` to enable them.')
+}
 
 module.exports = createServer
 
 function createServer (port, cb) {
+  if (!Server || !initDb) return cb(null)
```
Comment (ralphtheninja) [inline test/helpers/auth-server.js]: Maybe doesn't matter, but you're calling back synchronously and waking up zalgo :) Should really be

```js
if (!Server || !initDb) return process.nextTick(cb, null)
```

Comment (ralphtheninja) [inline test/helpers/auth-server.js]: Sorry for being nit picky :)

**Pull request merged**